src/grammar/cursor.rs

```diff
@@ -0,0 +1,483 @@
+use crate::{safe_loop, Parser};
+use source_gen::{lexer::TokenKind, syntax::SyntaxKind, T};
+
+use super::{parse_datatype, parse_expr, parse_ident, parse_query};
+
+pub fn parse_cursor(p: &mut Parser) {
```

Comment:
Where did you find a railroad diagram for this? Could you link it here in a comment?

Reply:
Will look into it

Reply:
for what exactly ?

Reply:
Do you mean the cursor in general? Here: https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/explicit-cursor-declaration-and-definition.html

Reply:
It's a bit hard to find because the structure of the docs is a bit confusing

Reply:
That's perfect, cheers. Just couldn't find it for some reason. Could you link it from the `parse_cursor` function?